examples/2016-10-31/lambda_edge/template.yaml

```diff
@@ -0,0 +1,79 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: Sample SAM configuration for Lambda@Edge to ease deployment and further updates
+Resources:
+
+    CFDistribution:
+      Type: AWS::CloudFront::Distribution
+      Properties:
+        DistributionConfig:
+          Enabled: 'true'
+          Comment: Lambda@Edge SAM Sample
+          DefaultRootObject: index.html
+          Origins:
+            -
+              Id: MyOrigin
+              DomainName: aws.amazon.com/lambda/edge/
+              CustomOriginConfig:
+                HTTPPort: 80
+                OriginProtocolPolicy: match-viewer
+          DefaultCacheBehavior:
+            TargetOriginId: MyOrigin
+            # Lambda@Edge configuration requires a function version not alias
+            LambdaFunctionAssociations:
+              - 
+                EventType: origin-request
+                # <SAM-Function.Version> provides {FunctionARN}:{Version} which is exactly what Cloudfront expects
+                # SAM Benefit here is upon function changes this function version will also be updated in Cloudfront
+                LambdaFunctionARN: !Ref LambdaEdgeFunctionSample.Version
+            ForwardedValues:
+              QueryString: 'false'
+              Headers:
+                - Origin
+              Cookies:
+                Forward: none
+            ViewerProtocolPolicy: allow-all
+
+    LambdaEdgeFunctionSample:
+        Type: AWS::Serverless::Function
+        Properties:
+          CodeUri: s3://<bucket>/lambda_edge.zip
+          Role: !GetAtt LambdaEdgeFunctionRole.Arn
```

Comment:
This is totally valid but unsure if you are aware that SAM can generate this role for you. If will give nearly the same role you have below (-edgelambda.amazonaws.com service). You can then append additional Policies to the generated Role through the `Policies` Property on the `AWS::Serverless::Function`. 

Personally, I like having SAM generate this resource. For this example, I don't think it matters as much but at least letting you know the other option in case you were not aware.

Reply:
Yep I'm aware of that and I'd have loved to use except for the Trust Policy :(. 

You can append Policies but you cannot append a Trust Policy - I tried all approaches before keeping on this one. This could've been possible if we allow customers to supply that perhaps through Globals or as an additional property.

If that were to work then I could simply use `LambdaEdgeFunctionSampleRole`.